Rakefile

```diff
@@ -215,6 +215,25 @@ namespace :dev do
     end
   end
 
+  task :gemfile
+  puts "\nGenerating Gemfile"
+  File.open('Gemfile', 'wb') do |file|
+    file.puts(<<~TEXT)
+      source "https://rubygems.org"
+      gemspec
+
+      gem 'colorator'
+      gem 'memory_profiler'
+      gem 'rake'
+    TEXT
+  end
+
+  desc "Profile memory usage from the test-cases"
+  task profile_memory: [:gemspec, :gemfile] do
+    puts ""
+    ruby 'test/profile_memory_usage.rb'
+  end
```

Comment:
Why would you need the `gemfile`? We only need to make sure that the `memory_profiler` and `colorator` gems are installed for this to work, a simple hint to stdout that one would need to run `gem install colorator memory_profile` in case of errors would be sufficient.